casbin/enforcer.h

```diff
@@ -147,9 +147,17 @@ class Enforcer : public IEnforcer{
         void BuildIncrementalRoleLinks(policy_op op, string p_type, vector<vector<string>> rules);
         // Enforce decides whether a "subject" can access a "object" with the operation "action", input parameters are usually: (sub, obj, act).
         bool Enforce(Scope scope);
+        // Enforce with two params, decides whether a "subject" can do the operation "action", input parameters are usually: (sub, act).
+        bool Enforce(string sub, string act);
+        // Enforce with three params, decides whether a "subject" can access a "object" with the operation "action", input parameters are usually: (sub, obj, act).
+        bool Enforce(string sub, string obj, string act);
+        // Enforce with four params, decides whether a "subject" can access a "object" with the operation "action" in the domain "dom", input parameters are usually: (sub, dom, obj,act).
+        bool Enforce(string sub, string dom, string obj, string act);
+        // Enforce with a vector param,decides whether a "subject" can access a "object" with the operation "action", input parameters are usually: (sub, obj, act).
+        bool Enforce(vector<string> params);

```

Comment:
@ZipoChan how about we use map or unordered_map, instead of vector as it would be convenient enough, so we won't have to run a loop through r_tokens and check for their names and substring it.
Like this,
`bool Enforce(map<string, string> params)`

Reply:
> ,



> @ZipoChan how about we use map or unordered_map, instead of vector as it would be convenient enough, so we won't have to run a loop through r_tokens and check for their names and substring it.
> Like this,
> `bool Enforce(map<string, string> params)`

Maybe we can overload a function like that,but we also need a function with vector,to be in accordance with the go version.In the go version,we don't need to pass rtokens,because the rtokens should be provided from the configuration file.

Reply:
I find that the Enforce(map<string,string> params) will conflict with Enforce(string sub,string act). 
I think there is no need to use map,because the go version doesn't use map.

Reply:
>I find that the Enforce(map<string,string> params) will conflict with Enforce(string sub,string act).
I think there is no need to use map,because the go version doesn't use map.

This can't be an answer to a question, because the Go core library also does not use vector or list either, but due to the language restrictions, we cannot support variadic arguments here.

Reply:
How will these two functions conflict with each other?
`Enforce(map<string,string> params)`
`Enforce(string sub,string act)`